cli/spinner.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+const encoder = new TextEncoder();
+
+const LINE_CLEAR = encoder.encode("\r\u001b[K"); // From cli/prompt_secret.ts
+const COLOR_RESET = "\u001b[0m";
+const DEFAULT_SPEED = 75;
+const DEFAULT_SPINNER = ["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"];
+
+/** Options for {@linkcode Spinner}. */
+export interface SpinnerOptions {
+  /**
+   * The sequence of characters to be iterated through for animation.
+   *
+   * @default {["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"]}
+   */
+  spinner?: string[];
+  /** The message to display next to the spinner. */
+  message?: string;
+  /** The speed of the spinner.
+   *
+   * @default {75}
+   */
+  speed?: number;
+  /** The color of the spinner. Defaults to the default terminal color. */
+  color?: string;
+  /**
+   * Whether to cleanup the output if the process is terminated.
+   *
+   * @default {true}
+   */
+  cleanup?: boolean;
```

Comment:
Hm... I wonder whether this option and the associated functionality are needed. I feel like it might not be. WDYT?

Reply:
After looking at it from the lens of someone using it, I don't see it getting used very much, and it being a hindrance to debugging when you stop a program for it to just clean up the output by default.